mpi_ring_c_test.sh

```diff
@@ -0,0 +1,80 @@
+#!/bin/bash
+
+function check_efa_ompi {
+    out=$1
+    grep -q "mtl:ofi:prov: efa" $out
+    if [ $? -ne 0 ]; then
+        echo "efa provider not used with Open MPI"
+        exit 1
+    fi
+}
+
+function check_efa_impi {
+    out=$1
+    grep -q "libfabric provider: efa" $out
+    if [ $? -ne 0 ]; then
+        echo "efa provider not used with Intel MPI"
+        exit 1
+    fi
+}
+
+function ompi_setup {
+    . /etc/profile.d/efa.sh
+    export OMPI_MCA_mtl_base_verbose=100
+}
+
+function impi_setup {
+    source $IMPI_ENV
+    export I_MPI_DEBUG=1
+}
+
+. ~/.bash_profile
```

Comment:
Any reason to use . vs source? I'm not very familiar with the difference, but I see that we do both in different files.

Reply:
They're equivalent and can be used interchangeability.